scheduler/src/cook/components.clj

```diff
@@ -231,10 +239,12 @@
                                    :request-header-size 32768})]
                       (fn [] (.stop jetty))))
 
-     :framework-id (fnk [curator-framework [:settings mesos-leader-path]]
-                     (when-let [bytes (curator/get-or-nil curator-framework
-                                                          (str mesos-leader-path "/framework-id"))]
-                       (String. bytes)))
+     :framework-id-promise (fnk [curator-framework [:settings mesos-leader-path]]
+                             (let [fid-promise (promise)]
+                               (when-let [bytes (curator/get-or-nil curator-framework
+                                                                    (str mesos-leader-path "/framework-id"))]
+                                 (deliver fid-promise (String. bytes)))
+                               fid-promise))
```

Comment:
Can we instead perform the blocking read on the promise here? That would avoid a lot of the downstream changes.